docker/Dockerfile

```diff
@@ -1,47 +1,9 @@
 from espressopp/buildenv:ubuntu
 
-ARG COVERAGE
-ARG XTC
-ARG CC
-ARG CXX
-ARG BUILD_UG
-
-#for coverage
-ENV CI true
-ENV TRAVIS true
-ARG TRAVIS_BRANCH
-ARG TRAVIS_JOB_NUMBER
-ARG TRAVIS_PULL_REQUEST 
-ARG TRAVIS_JOB_ID
-ARG TRAVIS_TAG
-ARG TRAVIS_REPO_SLUG
-ARG TRAVIS_COMMIT
-
-RUN rm -rf /home/espressopp/.ccache
-COPY espressopp /home/espressopp/espressopp
-COPY ccache/ /home/espressopp/.ccache
-USER root
-RUN chown -R espressopp:espressopp /home/espressopp/espressopp /home/espressopp/.ccache
-USER espressopp
-
-WORKDIR /home/espressopp/espressopp
-RUN mkdir build
-
-WORKDIR build
-RUN ccache -z
-RUN cmake -DCMAKE_INSTALL_PREFIX=/usr -DWITH_XTC=${XTC} ${COVERAGE:+-DUSE_GCOV=ON} ..
-RUN make VERBOSE=1 -j2 all ${BUILD_UG:+ug doc ug-pdf}
-RUN ccache -s
-RUN if [ -z ${COVERAGE} ] && [ -z $BUILD_UG ]; then make test CTEST_OUTPUT_ON_FAILURE=1 TEST_H5MD=ON; fi
-RUN make install DESTDIR=${PWD}
-RUN cd .. && if [ ${COVERAGE} ]; then \
-  if [ ${CC} = clang ]; then \
-    $HOME/.local/bin/codecov --gcov-exec "llvm-cov gcov" > /dev/null; \
-  else \
-    $HOME/.local/bin/codecov > /dev/null; \
-  fi; \
-fi && cd -
-USER root
-RUN make install
-USER espressopp
-WORKDIR ../examples
+RUN git clone https://github.com/espressopp/espressopp.git --depth 1
```

Comment:
This is wrong, as it would always build master instead of current commit.

Use the checkout action in docker workflow.

Reply:
This is the docker file, not the github actions file. I can link the sources from github actions into the docker container, yes, but why. Do you want to have a deployed docker container for every work in progress branch?!? I thought docker containers are meant for end users who do not want to build espressopp themselves? Furthermore, this will produce errors in forked repositories which do not have the secrets. I think I need some clarification what your containers are meant for.

Reply:
Well we want to test if a certain set of changes still builds, right? Otherwise we might merge a PR just to realize the docker build is not broken. (Same for the documentation below). One can set `push` to false for no push events.

https://github.com/votca/votca/blob/master/.github/workflows/docker-build.yml#L34

Reply:
Building, yes. But do you also want to deploy? In particular, this will make all builds fail on forked repositories where the secrets are not available. I think deploy should be restricted to the master branch.

Reply:
Yes, I would only deploy it for the main repo and pushed to master (and maybe nightly builds)